### PR TITLE
Add the OvenPlayer type declaration file for the first time

### PR DESCRIPTION
**Pull request by SangwonOh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
The api looks OK, but how it will be consumed? are you expecting users to use a global script version?
It seems the SDK is distributed as UMD package:
https://unpkg.com/ovenplayer@0.10.17/dist/ovenplayer.js
https://unpkg.com/browse/ovenplayer@0.10.17/package.json
(L:5) and there is `OvenPlayer` global symbol available when inccuded as script (e.g. when used as 'external' in webpack bundle output)

**Comment by SangwonOh** [comment]
@peterblazejewicz Hi. Thanks for reviewing :)
Because OvenPlayer is a UMD library, it can be used in a variety of ways. 
Do I need to add or change anything?
```html
<script src="https://cdn.jsdelivr.net/npm/ovenplayer/dist/ovenplayer.js"></script>

<script>
    // Initialize OvenPlayer
    const player = OvenPlayer.create('player_id', options)
</script>
```
or
```javascript
import OvenPlayer from 'ovenplayer';

// Initialize OvenPlayer
const player = OvenPlayer.create('player_id', options)
```

**Comment by peterblazejewicz** [comment]
This is how you can do this:
```diff
diff --git a/types/ovenplayer/index.d.ts b/types/ovenplayer/index.d.ts
index 162a82c517..28ced0c6c3 100644
--- a/types/ovenplayer/index.d.ts
+++ b/types/ovenplayer/index.d.ts
@@ -98,4 +98,6 @@ interface OvenPlayerInstance {
     off(eventName: string): void;
 }

-export default OvenPlayer;
+
+export as namespace OvenPlayer;
+export = OvenPlayer;
diff --git a/types/ovenplayer/ovenplayer-tests.ts b/types/ovenplayer/ovenplayer-tests.ts
index 3ecbc51e20..cdec54c6b4 100644
--- a/types/ovenplayer/ovenplayer-tests.ts
+++ b/types/ovenplayer/ovenplayer-tests.ts
@@ -1,4 +1,4 @@
-import OvenPlayer from 'ovenplayer';
+import OvenPlayer = require('ovenplayer');

 const player = OvenPlayer.create('player', {
     mute: true,
```
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html

**Comment by SangwonOh** [comment]
Thanks for guide. 

Because of the code below, it is not passing the `npm test ovenplayer`. I'll dig a little further, but do you know how to resolve `export-just-namespace` in our situation?
```
+export = OvenPlayer;
```
```
Error: C:/Users/admin/workspace/DefinitelyTyped/types/ovenplayer/index.d.ts:102:1
ERROR: 102:1  export-just-namespace  Instead of `export =`-ing a namespace, use the body of the namespace as the module body. See: https://github.com/Microsoft/dtslint/blob/master/docs/export-just-namespace.md
```

**Comment by peterblazejewicz** [comment]
Oh, yeas, there is no single symbol export in actual code.
I've played with this module:
```ts
// Import stylesheets
import OvenPlayer from 'ovenplayer';

// Write Javascript code!
const appDiv = document.getElementById('app');
OvenPlayer.debug(true);
var player = OvenPlayer.create(appDiv, {
  waterMark: {
    text: 'Text for water mark',
    font: {
      'font-size': '20px',
      color: '#fff',
      'font-weight': 'bold',
    },
  },
});
OvenPlayer.getPlayerList();
OvenPlayer.getPlayerByIndex(0);
OvenPlayer.getPlayerByContainerId('app');
```
you can try this pattern:
```ts
interface Source {
    ....
}

type PlayList = Source[][];

interface Config {
    ...
}

type CallbackFunction = (...args: any[]) => void;

interface OvenPlayerInstance {
    ...
}

export function create(container: string | HTMLDivElement, config: Config): OvenPlayerInstance;

export as namespace OvenPlayer;
```

Note that your code is missing  other functions (debug/getPlayerList/etc) from documentation

**Comment by SangwonOh** [comment]
@peterblazejewicz Yes. Thanks. All right. If I finish this modification, can I close this pull request and re-submit the pull request?
And I will add missing functions.

**Comment by peterblazejewicz** [comment]
sure, just reference this pr in the comments and you can ping me as well, thx!

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
Marking as needing revision per comment thread

**Comment by SangwonOh** [comment]
I just Merged branch 'DefinitelyTyped:master' into my master branch. No code changes yet.

**Review by gabritto — CHANGES_REQUESTED** [review]
waiting for the changes proposed in the comments, marking this as "request changes" so the bot pings a DT reviewer again when you make your changes :)

File: types/ovenplayer/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "target": "es5",
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/ovenplayer/tsconfig.json:11]:
```suggestion
```
you no longer needs this, your test file has proper import. Users can add this to their tsconfig (and by default TSC compiler add this in `--init` command):
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

File: types/ovenplayer/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "target": "es5",
```
Comment (peterblazejewicz) [inline types/ovenplayer/tsconfig.json:8]:
```suggestion
        "target": "es6",
```
that does not impact users code transpilation output, it's internal here for DT process. Keep this to default, unless there is a specific reason to change that value.

**Review by amcasey — CHANGES_REQUESTED** [review]
Please address the feedback from @peterblazejewicz

File: types/ovenplayer/tsconfig.json (hunk 2 above)
Comment (SangwonOh) [reply types/ovenplayer/tsconfig.json]: @peterblazejewicz  I got to know more about typescript. thank you.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@SangwonOh thx!

**Review by amcasey — APPROVED** [review]

**Comment by SangwonOh** [comment]
Ready to merge

**Pull request merged**